src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_BinaryOperator.cs

```diff
@@ -643,13 +645,14 @@ private BoundExpression UnconvertArrayLength(BoundArrayLength arrLength)
 
             if (tempAssignment != null)
             {
+                Debug.Assert(temp is { });
```

Comment:
>Debug.Assert(temp is { }); [](start = 16, length = 26)

nit: This is actually a case where suppression seems okay. But I don't assertion either